modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/DefaultDeclarationSessionBuilder.java

```diff
@@ -41,9 +44,14 @@ public DefaultDeclarationSessionBuilder(DefaultApplicationFactory defaultApplica
              String.valueOf(!valueOf(getProperty(MULE_FORCE_TOOLING_APP_LOGS_DEPLOYMENT_PROPERTY, "false"))))
         .put(MULE_LAZY_INIT_DEPLOYMENT_PROPERTY, TRUE)
         .put(MULE_LAZY_INIT_ENABLE_DSL_DECLARATION_VALIDATIONS_DEPLOYMENT_PROPERTY, TRUE)
+        .put(SHARED_PARTITIONED_PERSISTENT_OBJECT_STORE_PATH, getToolingWorkingDir().getAbsolutePath())
         .build();
   }
 
+  private File getToolingWorkingDir() {
+    return new File(getExecutionFolder(), "tooling");
```

Comment:
Does this work if "tooling" folder does not exist?

Reply:
Yes, it will create the directory.